File: src/main/java/com/example/dpop/orchestrator/api/v1/ChannelSessionResponse.java
package com.example.dpop.orchestrator.api.v1;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.example.dpop.orchestrator.session.ChannelState;

import java.util.List;
import java.util.UUID;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record ChannelSessionResponse(
        UUID channelSessionId,
        ChannelState state,
        String currentAcr,
        List<String> currentAmr,
        Boolean stepUpRequired,
        Long accountId
) {}

File: src/main/kotlin/com/example/dpop/orchestrator/api/v1/ChannelSessionResponse.kt
package com.example.dpop.orchestrator.api.v1

import com.fasterxml.jackson.annotation.JsonInclude
import com.example.dpop.orchestrator.session.ChannelState
import java.util.UUID

@JsonInclude(JsonInclude.Include.NON_NULL)
@JvmRecord
data class ChannelSessionResponse(
    val channelSessionId: UUID?,
    val state: ChannelState?,
    val currentAcr: String?,
    val currentAmr: List<String>?,
    val stepUpRequired: Boolean?,
    val accountId: Long?
)
